src/validators/user-info-request-validator.ts

```diff
@@ -15,26 +15,28 @@ export const userInfoRequestValidator = async (
       error: UserInfoRequestError;
     }
 > => {
-  const accessToken = userInfoRequestHeaders.authorization;
-  if (!accessToken) {
+  const authorisationHeader = userInfoRequestHeaders.authorization;
+  if (!authorisationHeader) {
     logger.warn("Missing authorisation header.");
     return { valid: false, error: UserInfoRequestError.MISSING_TOKEN };
   }
 
   let match;
   try {
-    match = /^Bearer (?<token>.*)$/.exec(accessToken);
+    match = /^Bearer (?<token>.*)$/.exec(authorisationHeader);
   } catch (error) {
     logger.error("Error parsing authorisation header.", error);
     return { valid: false, error: UserInfoRequestError.INVALID_TOKEN };
   }
```

Comment:
Not your code, but I think this isn't doing what it reads it's doing. If the `.exec` doesn't find the pattern, it returns null, as opposed to throwing an error. This isn't being caught because there's no test for this case. I think a null check rather than try-catch would be more appropriate.

```
const match = /^Bearer (?<token>.*)$/.exec(authorisationHeader);
if (!match) {
    logger.warn("Error parsing authorisation header.");
    return { valid: false, error: UserInfoRequestError.INVALID_TOKEN };
}
```
and perhaps a test if you have the time, but not essential:)

Reply:
Yeah I see what you mean, I'd favour just doing something like:
```
if(!authorisationHeader.includes("Bearer ")){
return error 
}

const accessToken = authorisationHeader.substring("Bearer ".length);
```

but then again I just don't like regex 😅

Reply:
That is quite neat actually, given the match is so simple. I like both, happy for you to choose which